install_utils.py

```diff
@@ -0,0 +1,172 @@
+    else:
+        raise ValueError("Invalid default answer: '%s'" % default)
+
+    while True:
+        sys.stdout.write(question + prompt)
+        choice = raw_input().lower()
+        if default is not None and choice == '':
+            return valid[default]
+        elif choice in valid:
+            return valid[choice]
+        else:
+            sys.stdout.write(
+                "Please respond with 'yes' or 'no' "
+                "(or 'y' or 'n').\n")
+
+
+def get_input(prompt, default=None):
+    """
+    Get user's input, only checking for non-empty response.
+    """
+    user_input = ""
+    if default is not None:
+        prompt = prompt+" (default: "+default+")"
+
+    while user_input == "":
+        user_input = raw_input(prompt+"\n")
+
+        if user_input == "":
+            if default is not None:
+                user_input = default
+            else:
+                print "The value cannot be blank."
+
+    return user_input
+
+
+# helper functions converted from one line script utils to python callable
+def print_warn(msg):
+    call_command("tput setaf 3")  # 3 = yellow
+    sys.stdout.write("[ WARNING ]\n")
+    call_command("tput sgr0")
+    sys.stderr.write(msg+"\n")
+
+
+def print_failure():
+    print ""
+    call_command("tput setaf 1")  # 1 = red
+    print_right("[ FAILED ]")
+    call_command("tput sgr0")
+
+
+def print_success():
+    call_command("tput setaf 2")  # 2 = green
+    print_right("[ OK ]")
+    call_command("tput sgr0")
+
+
+def print_step(msg):
+    call_command("tput setaf 6")
+    sys.stdout.write(msg+"\n")
+    call_command("tput sgr0")
+
+
+def print_right(msg):
+    call_command("tput cuu1")
+    call_command("tput cuf $(tput cols)")
+    call_command("tput cub %d" % len(msg))
+    sys.stdout.write(msg+"\n")
+
+
+def exit_with_message(msg):
+    sys.stderr.write(msg+"\n")
+    sys.exit(1)
+
+
+def exit_with_failrue(msg):
+    print_failure()
+    exit_with_message(msg)
+
+
+# utils using subprocess
+def call_command(command):
+    """
+    Process the given command in a bash shell and return the returncode.
+
+    Warning: Make sure the command is sanitized before
+    calling this function to prevent any vulnerability.
+    """
+    res = subprocess.call(
+        command, shell=True,
+        executable='/bin/bash')
+    return res
+
+
+def command_exists(command):
+    """From install.sh
+    """
+    res = call_command("hash " + command + " >/dev/null 2>&1")
+    if res != 0:
+        return False
+    else:
+        return True
+
+
+def get_command_output(command):
+    try:
+        res = subprocess.check_output(
+            command, shell=True,
+            stderr=subprocess.STDOUT,
+            executable='/bin/bash')
+    except:
+        res = None
+
+    return res
+
+
+# utils using os
+def check_path_exists(path, expand=False):
+    if(expand):
+        path = os.path.expanduser(path)
+
+    return os.path.exists(path)
+
+
+# Other helpers
+def write_file(filename):
```

Comment:
So far, I am only writing to file.  Is this method extraneous?

On Wed, Jul 6, 2016 at 1:03 PM, Jason Bau <notifications@github.com> wrote:

> In install_utils.py
> <https://github.com/wavefrontHQ/install/pull/22#discussion_r69801461>:
>
> > +    except:
> > +        res = None
> > +
> > +    return res
> > +
> > +
> > +# utils using os
> > +def check_path_exists(path, expand=False):
> > +    if(expand):
> > +        path = os.path.expanduser(path)
> > +
> > +    return os.path.exists(path)
> > +
> > +
> > +# Other helpers
> > +def write_file(filename):
>
> you're just opening/touching the file, right? maybe rename?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wavefrontHQ/install/pull/22/files/cd4dd95748662ae81a055276b6fdc466da95710a#r69801461>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALZQ32rb53DEnqb23wpKoD08sctha80nks5qTAoCgaJpZM4JFrB2>
> .
>